### PR TITLE
[R] Drop support for text inputs.

### PR DESCRIPTION
**Pull request by trivialfis**
Related: https://github.com/dmlc/xgboost/issues/9472

- Drop support for test inputs.
- Some minor fixes for document rendering.

Most changes are related to the document since we still need the URI API for binary input.

### CODE REVIEW

File: R-package/R/xgb.DMatrix.R [hunk 1]
```diff
@@ -9,12 +9,13 @@
...
 #' sorted-indices method (`tree_method = "exact"`), nor for the approximate method
 #' (`tree_method = "approx"`).
+#'
 #' @param data Data from which to create a DMatrix, which can then be used for fitting models or
 #' for getting predictions out of a fitted model.
 #'
 #' Supported input types are as follows:\itemize{
 #' \item `matrix` objects, with types `numeric`, `integer`, or `logical`.
-#' \item `data.frame` objects, with columns of types `numeric`, `integer`, `logical`, or `factor`.
+#' \item `data.frame` objects, with columns of types `numeric`, `integer`, `logical`, or `factor`}
```
Comment (david-cortes) [inline R-package/R/xgb.DMatrix.R:18]: Looks like we have a mixture of markdown ("-" for bullet points) and roxygen ("\itemize" for list and "\item" for each point).

Should ideally use only one format for a given function doc.

Comment (trivialfis) [reply R-package/R/xgb.DMatrix.R:18]: Will convert them

Comment (trivialfis) [reply R-package/R/xgb.DMatrix.R]: Done.

File: R-package/R/xgb.DMatrix.R [hunk 2]
```diff
@@ -23,33 +24,14 @@
...
-#'   }
+#' - CSR matrices, as class `dgRMatrix` from package `Matrix`.
+#' - CSC matrices, as class `dgCMatrix` from package `Matrix`.
 #'
-#' Be aware that the format of the file will not be auto-deduced - for example, if a file is named 'file.csv',
-#' it will not look at the extension or file contents to determine that it is a comma-separated value.
-#' Instead, the format must be specified following the URI format, so the input to `data` should be passed
-#' like this: `"file.csv?format=csv"` (or `"file.csv?format=csv&label_column=0"` if the first column
-#' corresponds to the labels).
+#' These are **not** supported by the `xgb.QuantileDMatrix`.
```
Comment (david-cortes) [inline R-package/R/xgb.DMatrix.R:30]:
```suggestion
#' These are **not** supported by `xgb.QuantileDMatrix`.
```

**Review by david-cortes — APPROVED** [review]

**Pull request merged**